### PR TITLE
Add change list-grid view test

### PR DESCRIPTION
**Pull request by eatonphil**

### CODE REVIEW

File: test/linodes/layouts/IndexPage.spec.js [hunk 1]
```diff
@@ -185,5 +185,30 @@ describe('linodes/layouts/IndexPage', () => {
...
+          dispatch={localDispatch}
+          view={initial}
+          selected={{}}
+          linodes={linodes}
+        />
+      );
+
+      const listButton = page.find(`.grid-list .${final}`);
+      listButton.simulate('click');
+      expect(viewChangeType).to.have.property(final);
```
Comment (SirCmpwn) [inline test/linodes/layouts/IndexPage.spec.js]: Eeek I don't like this test. Don't make a magic dispatch and mutate a variable in some shared scope, just assert that dispatch gets called with the appropriate CHANGE_VIEW action via `dispatch.calledWith({ type: CHANGE_VIEW, view: 'grid' })`

Comment (eatonphil) [inline test/linodes/layouts/IndexPage.spec.js]: Ah yes, that was silly. Thanks for the smack on the head.

**Comment by SirCmpwn** [comment]
Ready?

**Comment by eatonphil** [comment]
Yep.

**Comment by eatonphil** [comment]
thanks!

**Pull request merged**